fx quant: fix types on _find_quants

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49717 fx quant: hook up ConvTranspose{n}d
* #49688 fx quant: improve types on convert
* #49628 fx quant: do not observe bias on F.linear
* #49623 fx quant: do not observe bias on F.conv
* **#49616 fx quant: fix types on _find_quants**
* #49607 fx quant: add types to observed_module.py
* #49606 fx quant: types for fusion_patterns.py
* #49549 unbreak mypy torch/quantization

Summary:

Add types to `_find_quants` I/O and fix resulting errors,
needed for an upcoming bug fix.

Test Plan:

```
mypy torch/quantization
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25645719](https://our.internmc.facebook.com/intern/diff/D25645719)

vkuzo committed:
fx quant: fix types on _find_quants

Summary:

Add types to `_find_quants` I/O and fix resulting errors,
needed for an upcoming bug fix.

Test Plan:

```
mypy torch/quantization
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -981,7 +985,7 @@ def visit_arg(arg):
                     # necessary
                     quants[arg.name] = (
-                        DefaultQuantizeHandler(self, arg), act_post_process_ctr)
+                        DefaultQuantizeHandler(self, arg), act_post_process_ctr, is_weight)
```
I think we don't need is_weight, this info is used to decide act_post_process_ctr already in L976

jerryzh168 replied:
also we can remove L978-979 I think

jerryzh168 commented on torch/quantization/fx/quantize.py:
```diff
@@ -956,7 +959,8 @@ def _find_quants(self, graph: Graph, matches: Dict[str, MatchResult],
         """
-        quants: Dict[str, Any] = {}
+        # quants: Dict[str, Any] = {}
+        quants: Dict[str, Tuple[DefaultQuantizeHandler, Callable, bool]] = {}
```
we can remove `bool` here

jerryzh168 reviewed (changes requested):
as discussed in chat, we can remove `DefaultQuantizeHandler`, thanks!

vkuzo committed:
Update on "fx quant: fix types on _find_quants"


Summary:

Add types to `_find_quants` I/O and fix resulting errors,
needed for an upcoming bug fix.

Test Plan:

```
mypy torch/quantization
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25645719](https://our.internmc.facebook.com/intern/diff/D25645719)

[ghstack-poisoned]

vkuzo commented on torch/quantization/fx/quantize.py:
```diff
@@ -571,14 +574,9 @@ def load_non_quantized(n):
             return env[n.name]
 
         def load_quantized(n):
-            if n.name not in quant_env:
```
this code was flagged unreachable, deleting

vkuzo committed:
nit on "fx quant: fix types on _find_quants"


Summary:

Add types to `_find_quants` I/O and fix resulting errors,
needed for an upcoming bug fix.

Test Plan:

```
mypy torch/quantization
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25645719](https://our.internmc.facebook.com/intern/diff/D25645719)

[ghstack-poisoned]

vkuzo commented:
> as discussed in chat, we can remove DefaultQuantizeHandler, thanks!

separate PR, to keep things small? since this is really just annotating types and fixing bugs uncovered by that

vkuzo committed:
Update on "fx quant: fix types on _find_quants"


Summary:

Add types to `_find_quants` I/O and fix resulting errors,
needed for an upcoming bug fix.

Test Plan:

```
mypy torch/quantization
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25645719](https://our.internmc.facebook.com/intern/diff/D25645719)

[ghstack-poisoned]

vkuzo committed:
Update on "fx quant: fix types on _find_quants"


Summary:

Add types to `_find_quants` I/O and fix resulting errors,
needed for an upcoming bug fix.

Test Plan:

```
mypy torch/quantization
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25645719](https://our.internmc.facebook.com/intern/diff/D25645719)

[ghstack-poisoned]
